splunk_handler/__init__.py

```diff
@@ -126,16 +121,18 @@ def emit(self, record):
             self.write_log(traceback.format_exc())
             return
 
-        if self.flush_interval > 0:
-            try:
-                self.write_debug_log("Writing record to log queue")
-                # Put log message into queue; worker thread will pick up
-                self.queue.put_nowait(record)
-            except Full:
-                self.write_log("Log queue full; log data will be dropped.")
-        else:
+        if self.flush_interval <= 0:
             # Flush log immediately; is blocking call
             self._splunk_worker(payload=record)
+            return
+
+        self.write_debug_log("Writing record to log queue")
+        # Put log message into queue; worker thread will pick up
+        if self.cur_queue_size < self.max_queue_size:
+            self.queue.append(record)
+            self.cur_queue_size += 1
+        else:
+            self.write_log("Log queue full; log data will be dropped.")
```

Comment:
This seems like something that raise an error rather than silently printing a message. That would allow the logging application to handle the issue.

Thoughts?

Reply:
I think that's a good change, though is potentially breaking. Let's do that as a follow-up and decide if it would warrant a major version bump.